src/tree/updater_gpu_hist.cu

```diff
@@ -809,12 +810,11 @@ class GPUHistMaker : public TreeUpdater {
     collective::Broadcast(&column_sampling_seed, sizeof(column_sampling_seed), 0);
 
     auto batch_param = BatchParam{param->max_bin, TrainParam::DftSparseThreshold()};
-    auto page = (*dmat->GetBatches<EllpackPage>(ctx_, batch_param).begin()).Impl();
```

Comment:
This initiates an iteration on the sparse DMatrix but doesn't finish it. As a result, we run sketching twice before the PR.